grails-app/controllers/org/pih/warehouse/user/DashboardController.groovy

```diff
@@ -198,9 +198,16 @@ class DashboardController {
             // Save the warehouse selection for "last logged into" information
             if (user) {
                 //userInstance.rememberLastLocation = Boolean.valueOf(params.rememberLastLocation)
-                user.lastLoginDate = new Date()
-                user.save(flush: true)
-                session.user = user
+
+                /*
+                 * Band-aid for OBGM-355, to prevent the following exception at dashboard launch:
+                 * javax.persistence.TransactionRequiredException: no transaction is in progress
+                 */
+                user.withTransaction {
```

Comment:
This doesn't seem right. First of all, unless I;m I think it should be `User.withTransaction { }`. Secondly, we just need to annotate the method to be @Transactional.

Reply:
You're right on both counts, I think -- will fix